src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/Type.cs

```diff
@@ -585,16 +585,6 @@ public PredefinedType getPredefType()
             return getAggregate().GetPredefType();
         }
 
-        ////////////////////////////////////////////////////////////////////////////////
-        // Is this type System.TypedReference or System.ArgIterator?
-        // (used for errors because these types can't go certain places)
-
-        public bool isSpecialByRefType()
-        {
-            // ArgIterator, TypedReference and RuntimeArgumentHandle are not supported.
```

Comment:
If they were going to be supported by dynamic (4.6 has these types, but doesn't support them in dynamic AFAICT), it would be necessary to put this back, but it would also be necessary to create a `CType`-derived class for whichever of these was allowed along with code handling those, and have either new binder types and/or somehow change the way you can't assign a `TypedReference` to `object` etc.

Bringing this back would be the easiest part of this.